src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/PackageLibs.targets

```diff
@@ -3,10 +3,47 @@
   <PropertyGroup>
     <PackagingTaskDir Condition="'$(PackagingTaskDir)' == ''">$(MSBuildThisFileDirectory)</PackagingTaskDir>
     <GenerationDefinitionFile Condition="'$(GenerationDefinitionFile)' == ''">$(MSBuildThisFileDirectory)Generations.json</GenerationDefinitionFile>
+    <RuntimeIdGraphDefinitionFile Condition="'$(RuntimeIdGraphDefinitionFile)' == '' AND Exists('$(ProjectDir)pkg/Microsoft.NETCore.Platforms/runtime.json')">$(ProjectDir)pkg/Microsoft.NETCore.Platforms/runtime.json</RuntimeIdGraphDefinitionFile>
+    <RuntimeIdGraphDefinitionFile Condition="'$(RuntimeIdGraphDefinitionFile)' == ''">$(MSBuildThisFileDirectory)runtime.json</RuntimeIdGraphDefinitionFile>
   </PropertyGroup>
 
   <UsingTask TaskName="ValidatePackageTargetFramework" AssemblyFile="$(PackagingTaskDir)Microsoft.DotNet.Build.Tasks.Packaging.dll"/>
+  <UsingTask TaskName="GetApplicableAssetsFromPackages" AssemblyFile="$(PackagingTaskDir)Microsoft.DotNet.Build.Tasks.Packaging.dll"/>
 
+  <Target Name="UpdatePkgProjProjectReferences" BeforeTargets="BeforeResolveReferences">
+    <!-- Update PkgProj references to call the GetValidationReport target -->
+    <ItemGroup>
+      <ProjectReference Condition="'%(Extension)'=='.pkgproj'">
+        <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+        <Targets>Build;GetPackageAssets</Targets>
+        <OutputItemType>PackageAsset</OutputItemType>
+      </ProjectReference>
+    </ItemGroup>
+  </Target>
+  
+  <PropertyGroup>
+    <!-- Resolve from pkgProjs before nuget packages -->
+    <ResolveNugetPackagesDependsOn>$(ResolveNugetPackagesDependsOn);ResolvePkgProjReferences</ResolveNugetPackagesDependsOn>
+    <ResolvePkgProjReferencesDependsOn>ResolveProjectReferences</ResolvePkgProjReferencesDependsOn>
+  </PropertyGroup>
+  
+  <Target Name="ResolvePkgProjReferences" Condition="'@(PackageAsset)' != ''" DependsOnTargets="$(ResolvePkgProjReferencesDependsOn)">
+    <PropertyGroup>
+      <!-- Temporary workaround -->
+      <NuGetTargetMoniker Condition="'$(NuGetTargetMoniker)' == '.NETPortable,Version=v4.5,Profile=Profile7'">DNXCore,Version=v5.0</NuGetTargetMoniker>
+    </PropertyGroup>
+    
+    <!-- Hack this for now to use TestNugetRuntimeId.  We probably want to run this in a way to get the compile assets for the
+         target platform and runtime assets for the RID-specific platform.  -->
+    <GetApplicableAssetsFromPackages PackageAssets="@(PackageAsset)" 
+                                     TargetMoniker="$(NuGetTargetMoniker)"
+                                     RuntimeTargetMoniker="$(TestNuGetTargetMoniker)"
+                                     TargetRuntime="$(TestNugetRuntimeId)"
```

Comment:
We should chat more about it but I do think we should have TestRID as well as TestTFM for consistency and the reason we want the shorter one is because I expect that people will need to specify it in their builds files. If that isn't the case any longer then we can consider not changing this. My main concern is making a connection between the different Test* properties so it is consistent and clear they both effect what we run on.